tests/e2e/default/targets/target-aggregates.wdio-spec.js

```diff
@@ -0,0 +1,420 @@
+        }
+      };
+
+      const targetDocs = _.flatten(docs
+        .filter(doc => doc.type === 'person')
+        .map(contact => {
+          const genTarget = (target) => {
+            const value = targetValuesByContact[contact.name] &&
+              targetValuesByContact[contact.name][target.id].value ||
+              { pass: randomNumber(100), total: randomNumber(100) };
+            return { id: target.id, value };
+          };
+
+          return docTags.map(tag => ({
+            _id: `target~${tag}~${contact._id}~irrelevant`,
+            reporting_period: tag,
+            targets: targetsConfig.map(genTarget),
+            owner: contact._id,
+            user: 'irrelevant',
+          }));
+        }));
+      await utils.saveDocs(targetDocs);
+      await updateSettings(targetsConfig, user);
+
+      await commonPage.goToAnalytics();
+      await targetAggregatesPage.goToTargetAggregates(true);
+
+      const expectedTargets = [
+        { id: 'count_no_goal', title: 'count no goal', progressBar: false, goal: false, counter: '27' },
+        { id: 'count_with_goal', title: 'count with goal', progressBar: true, goal: 20, counter: '3 of 5' },
+        { id: 'percent_no_goal', title: 'percent no goal', progressBar: true, goal: false, counter: '69%' },
+        { id: 'percent_with_goal', title: 'percent with goal', progressBar: true, goal: '80%', counter: '1 of 5' },
+        { id: 'percent_achieved', title: 'percent achieved', progressBar: true, goal: '10%', counter: '5 of 5' },
+      ];
+
+      const contacts = docs.filter(doc => doc.type === 'person' && doc.parent.parent._id === parentPlace._id);
+
+      await expectTargets(expectedTargets);
+      for (const target of expectedTargets) {
+        await targetAggregatesPage.openTargetDetails(target.id);
+        await targetAggregatesPage.expectTargetDetails(target);
+        const expectedContacts = contacts.map(contact => ({
+          _id: contact._id,
+          name: contact.name,
+          counter: targetValuesByContact[contact.name][target.id].counter,
+          progress: targetValuesByContact[contact.name][target.id].progress,
+        }));
+        await expectContacts(expectedContacts, target);
+      }
+
+      // refreshing with an open target works correctly
+      const target = expectedTargets[2];
+      await targetAggregatesPage.openTargetDetails(target.id);
+      await browser.refresh();
+      await targetAggregatesPage.expectTargetDetails(target);
+    });
+
+    it('should route to contact-detail on list item click and display contact summary target card', async () => {
+      const targetsConfig = [
+        { id: 'a_target', type: 'count', title: generateTitle('what a target!'), aggregate: true },
+        { id: 'b_target', type: 'percent', title: generateTitle('the most target'), aggregate: true },
+      ];
+      const contactSummaryScript = `
```

Comment:
It would help to move this script to a separate JavaScript file and request it like this. 
```
const contactSummaryScript = fs.readFileSync(`${__dirname}/config/contact-summary-target-aggregates.js`, 'utf8');
```
If I remember correctly, that already returns a string.